### PR TITLE
First commit for NEON ISA for `DD`

### PR DESCRIPTION
**Pull request by manodeep**
Both `rpavg` and `weightavg` are **wrong** and the tests fail locally. The good news is that the number of pairs match

* Edit *

Added a checklist for the PR

- [x] Corrfunc compiles and works correctly on ARM64 platforms with the fallback kernels
- [ ] Add ARM64 kernels for all theory routines
    - [x] DD
    - [ ] DDrppi
    - [ ] DDsmu
    - [ ] wp
    - [ ] xi
    - [ ] vpf
- [ ] Add ARM64 kernels for all mocks routines
    - [ ] DDrppi_mocks
    - [ ] DDsmu_mocks
    - [ ] DDtheta_mocks
    - [ ] vpf_mocks
- [ ] Add tests for all ARM64 kernels
- [ ] Check that the kernels compile and work correctly when installed through python
- [ ] Add notes in the documentation for ARM64 functionality
- [ ] Create new minor Corrfunc release

### CODE REVIEW

**Comment by manodeep** [comment]
It's a bit of a mixed bag. In single-precision - 3.6s with `NEON` and 5.8s with `FALLBACK`, while the times are about the same in double-precision.

**Comment by manodeep** [comment]
Realised that we have run out of Travis CI build minutes and *no* tests have run on TravisCI for nearly a year now. Hence, the `arm64` tests will have to be done on some other CI -> currently trying out CircleCI

File: theory/DD/countpairs_kernels.c.src [hunk 1]
```diff
@@ -19,6 +19,281 @@
...
+
+          NEON_INTS m_rpbin = NEON_SET_INT(0);
+          for(int kbin=nbin-1;kbin>=1;kbin--) {
+              const NEON_UINTS m1 = NEON_COMPARE_FLOATS_GE(r2,m_rupp_sqr[kbin-1]);
+              const NEON_UINTS m_bin_mask = NEON_BITWISE_AND(m1,m_mask_left);
+              m_mask_left = NEON_COMPARE_FLOATS_LT(r2,m_rupp_sqr[kbin-1]);
+              const int test2  = NEON_TEST_COMPARISON(m_bin_mask);
+              npairs[kbin] += NEON_BIT_COUNT_INT(test2);
+              if(need_rpavg || need_weightavg){
+                  m_rpbin = NEON_BLEND_INTS_WITH_MASK(m_rpbin, NEON_SET_INT(kbin), m_bin_mask);
```
Comment (lgarrison) [inline theory/DD/countpairs_kernels.c.src:216]: Everything looks good to me; I'm not seeing why `rpavg` and `weightavg` would be wrong. This line stands out as one place things could go wrong, but reading the documentation of these instructions, it looks okay to me.

**Comment by manodeep** [comment]
I am trying out a bunch of new optimisations along side this ARM64 implementation - might take a while to finalise. Will re-open once ready

**Pull request closed without merging**